src/components/validation/validators/component_spec/mod.rs

```diff
@@ -126,11 +126,24 @@ fn validate_telemetry(
     }
 }
 
-fn filter_events_by_metric_and_component<'a>(
+fn filter_events_by_metric_and_component_with_errors<'a>(
```

Comment:
nit: It is mildly confusing to me that this function name is "_with_errors" , but is called from the validation of metrics that are not error metrics, whereas the non "_with_errors" one is called from the one for error metrics.

Reply:
We need the inverse relationship on the validation side since we're testing for the absence of errors in the error validations. I'm not sure if there's a great way to express this beyond these names.